src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -45,29 +93,39 @@ public class PersistentShard : Shard
         /// <param name="extractShardId">TBD</param>
         /// <param name="handOffStopMessage">TBD</param>
         public PersistentShard(
+            IActorContext context,
+            IPersistentActorContext persistentContext,
+            Action<object> unhandled,
```

Comment:
This is similar to how AtLeastOnceDelivery is implemented.
If you have better idea how to get around scala traits, without code duplication, i'm open to suggestions.